src/services/web3/web3.service.ts

```diff
@@ -25,6 +45,59 @@ export default class Web3Service {
       ens: await this.getEnsName(address)
     };
   }
+
+  async getUserAddress(): Promise<string> {
+    const signer = this.provider.value.getSigner();
+    const userAddress: string = await signer.getAddress();
+    return userAddress;
+  }
```

Comment:
This is a little exception, we are reading data here but it's the userProvider address we're interested in.
```suggestion
  async getUserAddress(): Promise<string> {
    const signer = this.userProvider.value.getSigner();
    const userAddress: string = await signer.getAddress();
    return userAddress;
  }
```